### DISCUSSION TITLE
Why function composition doesn't work in the intuitive way?

### DISCUSSION THREAD

**Discussion by YonatanNemtsov** [sympy/sympy#26622] [General]
Hi, I would like function composition to be as simple as this:

```python
>>>theta = smp.symbols('theta', cls=smp.Function)
>>>x = smp.cos(theta1)
```

`x` would be of type Function in this case.
Is there a good reason why this shouldn't work in this way?

**Comment by asmeurer** [comment]
Syntactic questions aside, there's a technical reason this can't work right now https://github.com/sympy/sympy/issues/4787

**Reply by YonatanNemtsov** [reply]
Interesting, thank you. Is there work being done to change this in the future, or is there interest in it?
